### PR TITLE
cephadm: support quotes around public/cluster network in config passed to bootstrap

### PR DESCRIPTION
**Pull request by adk3798**
Fixes: https://tracker.ceph.com/issues/56973

Signed-off-by: Adam King <adking@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-07-30_00:56:25-orch:cephadm-wip-adk-testing-2022-07-29-1905-distro-default-smithi/

green

**Comment by fmount** [comment]
> @fmount pinging you on this as discussed
LGTM.

ack thanks

File: src/cephadm/tests/test_cephadm.py [hunk 1]
```diff
@@ -2404,6 +2404,41 @@ def test_ip_in_subnet(self):
         with pytest.raises(Exception):
             rc = cd.ip_in_sublets('fe80:2030:31:24', 'fe80::/64')
 
+    @pytest.mark.parametrize("conf", [
+    """[global]
+public_network='1.1.1.0/24,2.2.2.0/24'
+cluster_network="3.3.3.0/24, 4.4.4.0/24
```
Comment (phlogistonjohn) [inline src/cephadm/tests/test_cephadm.py:2410]: I sort of wish that unmatched quotes could still be an error (unless you tell me it wasn't in the old version), but I won't let it bother me too much.

**Review by phlogistonjohn — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-08-03_01:34:50-orch:cephadm-wip-adk-testing-2022-08-02-1725-distro-default-smithi/

a handful of infra/teuthology issues. all passed in rerun https://pulpito.ceph.com/adking-2022-08-03_12:58:27-orch:cephadm-wip-adk-testing-2022-08-02-1725-distro-default-smithi/

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-08-09_15:27:00-orch:cephadm-wip-adk-testing-2022-08-09-0854-distro-default-smithi/

failures due to infra issues with rhel tests. All passed in rerun
https://pulpito.ceph.com/adking-2022-08-09_20:24:21-orch:cephadm-wip-adk-testing-2022-08-09-0854-distro-default-smithi/

File: src/cephadm/tests/test_cephadm.py (hunk 1 above)
Comment (adk3798) [reply src/cephadm/tests/test_cephadm.py]: while it technically would work if someone provided this with the given code, I definitely did not intend to purposely support this and have now removed this test case (by adding the ending quote that should have been there)

**Pull request merged**